modelarmor/api/ModelArmor.Samples.Tests/ModelArmorFixture.cs

```diff
@@ -15,53 +15,180 @@
+    private const string EnvLocation = "GOOGLE_CLOUD_LOCATION";
+
     public ModelArmorClient Client { get; }
+    public DlpServiceClient DlpClient { get; }
     public string ProjectId { get; }
     public string LocationId { get; }
-    public TemplateName TemplateForQuickstartName { get; }
+    private readonly List<TemplateName> _maTemplatesToCleanup = new List<TemplateName>();
+    private readonly List<string> _dlpTemplatesToCleanup = new List<string>();
 
     public ModelArmorFixture()
     {
-        // Get the Google Cloud Project ID.
-        ProjectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
-        if (string.IsNullOrEmpty(ProjectId))
+        ProjectId = GetRequiredEnvVar(EnvProjectId);
+        LocationId = Environment.GetEnvironmentVariable(EnvLocation) ?? "us-central1";
+
+        // Create the Model Armor client.
+        ModelArmorClientBuilder clientBuilder = new ModelArmorClientBuilder
+        {
+            Endpoint = $"modelarmor.{LocationId}.rep.googleapis.com",
+        };
+        Client = clientBuilder.Build();
+
+        // Create the DLP client.
+        DlpClient = new DlpServiceClientBuilder { Endpoint = $"dlp.googleapis.com" }.Build();
+    }
+
+    private string GetRequiredEnvVar(string name)
+    {
+        var value = Environment.GetEnvironmentVariable(name);
+        if (string.IsNullOrEmpty(value))
         {
-            throw new Exception("Missing GOOGLE_PROJECT_ID environment variable");
+            throw new Exception($"Missing {name} environment variable");
         }
+        return value;
+    }
 
-        // Get location ID from environment variable or use default.
-        LocationId = Environment.GetEnvironmentVariable("GOOGLE_CLOUD_LOCATION") ?? "us-central1";
+    public string GenerateUniqueId()
+    {
+        return Guid.NewGuid().ToString("N").Substring(0, 8);
+    }
 
-        // Create client.
-        Client = ModelArmorClient.Create();
+    // Creates a DLP Inspect Template and returns its name.
+    public string CreateInspectTemplate(string displayName = "Test Inspect Template")
+    {
+        var parent = new LocationName(ProjectId, LocationId).ToString();
+        var templateId = $"inspect-{GenerateUniqueId()}";
+        var request = new CreateInspectTemplateRequest
+        {
+            Parent = parent,
+            InspectTemplate = new InspectTemplate
+            {
+                DisplayName = displayName,
+                InspectConfig = new InspectConfig
+                {
+                    InfoTypes =
+                    {
+                        new InfoType { Name = "PERSON_NAME" },
+                        new InfoType { Name = "EMAIL_ADDRESS" },
+                        new InfoType { Name = "US_INDIVIDUAL_TAXPAYER_IDENTIFICATION_NUMBER" },
+                    },
+                },
+            },
+            TemplateId = templateId,
+        };
+        var response = DlpClient.CreateInspectTemplate(request);
```

Comment:
Consider adding these to the clean up resource collections right here so you don't have to do it in all tests.